src/ray/core_worker/core_worker.cc

```diff
@@ -1324,6 +1326,16 @@ Status CoreWorker::CreateExisting(const std::shared_ptr<Buffer> &metadata,
   }
 }
 
+Status CoreWorker::ExperimentalMutableObjectWriteAcquire(
+    const ObjectID &object_id,
+    const std::shared_ptr<Buffer> &metadata,
+    uint64_t data_size,
+    int64_t num_readers,
```

Comment:
It it possible to have `num_readers` only passed in the initial create and not in subsequent operations?

Reply:
Hmm we can do this, but I think it's actually better to have the flexibility here. We can always make it an attr on the Python object instead of C++.